packages/alchemy/src/__tests__/simple-account.test.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  SimpleSmartContractAccount,
+  type BatchUserOperationCallData,
+  type SimpleSmartAccountOwner,
+} from "@alchemy/aa-core";
+import { toHex } from "viem";
+import { mnemonicToAccount } from "viem/accounts";
+import { polygonMumbai } from "viem/chains";
+import { AlchemyProvider } from "../provider";
+
+const ENTRYPOINT_ADDRESS = "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789";
+const API_KEY = process.env.API_KEY!;
+const OWNER_MNEMONIC = process.env.OWNER_MNEMONIC!;
+const PAYMASTER_POLICY_ID = process.env.PAYMASTER_POLICY_ID!;
+const SIMPLE_ACCOUNT_FACTORY_ADDRESS =
+  "0x9406Cc6185a346906296840746125a0E44976454";
+
+describe("Simple Account Tests", () => {
+  const ownerAccount = mnemonicToAccount(OWNER_MNEMONIC);
+  const owner: SimpleSmartAccountOwner = {
+    signMessage: async (msg) =>
+      ownerAccount.signMessage({
+        message: { raw: toHex(msg) },
+      }),
+    getAddress: async () => ownerAccount.address,
+  };
+  const chain = polygonMumbai;
+  const signer = new AlchemyProvider({
+    apiKey: API_KEY,
+    chain,
+    entryPointAddress: ENTRYPOINT_ADDRESS,
+  }).connect(
+    (provider) =>
+      new SimpleSmartContractAccount({
+        entryPointAddress: ENTRYPOINT_ADDRESS,
+        chain,
+        owner,
+        factoryAddress: SIMPLE_ACCOUNT_FACTORY_ADDRESS,
+        rpcClient: provider,
+      })
+  );
+
+  it("should succesfully get counterfactual address", async () => {
+    expect(await signer.getAddress()).toMatchInlineSnapshot(
+      `"0xb856DBD4fA1A79a46D426f537455e7d3E79ab7c4"`
+    );
+  });
+
+  it("should correctly sign the message", async () => {
+    expect(
+      // TODO: expose sign message on the provider too
+      await signer.account.signMessage(
+        "0xa70d0af2ebb03a44dcd0714a8724f622e3ab876d0aa312f0ee04823285d6fb1b"
+      )
+    ).toBe(
+      "0xd16f93b584fbfdc03a5ee85914a1f29aa35c44fea5144c387ee1040a3c1678252bf323b7e9c3e9b4dfd91cca841fc522f4d3160a1e803f2bf14eb5fa037aae4a1b"
```

Comment:
Does this depends on the specific value of the private key in the env var? If so, we could use a test key for this purpose (like the output of a string's hash, similar to foundry's [`makeAddrAndKey`](https://github.com/foundry-rs/forge-std/blob/66bf4e2c92cf507531599845e8d5a08cc2e3b5bb/src/StdCheats.sol#L429-L433)).

Reply:
yea it does depend on a specific key in the env var.

I can make that change to test signing, but will still need some live key with funds in it for testing the send user op methods